azure.yaml

```diff
@@ -1,17 +1,8 @@
-name: chat-rag-openai-csharp-prompty
+name: contoso-chat-csharp-prompty
 metadata:
-  template: chat-rag-openai-csharp-prompty@0.0.1
-hooks:
-  postprovision:
-    posix:
-      shell: sh
-      continueOnError: false
-      interactive: true
-      run: infra/hooks/postprovision.sh
-    windows:
-      shell: pwsh
-      continueOnError: false
-      interactive: true
-      run: infra/hooks/postprovision.ps1
-infra:
-    provider: "bicep"
\ No newline at end of file
+  template: contoso-chat-csharp-prompty@0.0.1
+services:
+  ContosoChatAPI:
+    project: ./src/ContosoChatAPI
+    image: "${AZURE_CONTAINER_REGISTRY_NAME}.azurecr.io/contosochatapi:latest"
```

Comment:
I am not sure this is correct, but I might be confused.
@wbreza , does `image` means:  "use this image instead of building any project?"   If yes, then this is not what we want here, do we?
We want azd to build a new image from the dockerfile in `./src/ContosoChatAPI/ContosoChatAPI/dockerfile` and the use that image to publish to containerapp.

Reply:
I know that @wbreza did add support for services backed by existing images.  I do agree with @vhvb1989 that this likely does too much - I think something needs to continue to build and push the container image.

The post provision scripts were doing two things before:

1. Using `az acr build` to queue a remote build of the image (so that customers didn't need to have Docker Desktop installed)
2. Using a combination of `az containerapp update` and `az containerapp ingress update` to create a new revision and configure ingress with it.

While we don't have a solution for (1) in `azd` today (using remote build) I do think the changes here allow us to remove (2) from the post provision scripts.

I think we'll need to continue to have the custom script to do the `az acr build` to queue the remote build until we can teach `azd` how to do that.  Perhaps we can move this into a pre-deploy script for the `ContosoChatAPI` service?

Reply:
Thank you, making these changes today.

Reply:
Agree that we shouldn't need to specify `image` here.  This is used when specifying source docker image when we are not building from source.

If you want to control the generated repo/image/tag format you can use the `docker` options to specify that otherwise it would use `azd` defaults which should be more than fine for this example use case.

Reply:
`azd` will automatically use the container registry set in `AZURE_CONTAINER_REGISTRY_NAME` environment variable.